### ISSUE TITLE
Calendar selecting wrong day on keyboard nav

### ISSUE THREAD

**Issue by Fercas123** [jpmorganchase/salt-ds#3340]
### Package name(s)

Lab (@salt-ds/lab)

### Description

keyboard navigation takes you to the first day, even when there is a selected day. order of priority should be: 
- selected
- today if the cal is in current month
- 1rst day

### Steps to reproduce

any calendar example

### Operating system

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Browser

- [X] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

### Are you a JPMorgan Chase & Co. employee?

- [X] I am an employee of JPMorgan Chase & Co.
